lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -427,6 +433,20 @@ class ModuleDiscriminator {
   /// Adds a string parameter.
   void addString(const Twine &name, const Twine &txt) {
     addParam(name, StringAttr::get(ctx, txt));
+
+    // Replace all non-alphanumeric characters by an underscore.
+    std::string cleanedName = txt.str();
+    std::replace_if(
+        cleanedName.begin(), cleanedName.end(),
+        [](char c) { return !std::isalnum(c); }, '_');
+
+    if (modName) {
+      cleanedName = *modName + "_" + cleanedName;
+    } else {
+      cleanedName = getOpName() + "_" + cleanedName;
+    }
+
+    modName = cleanedName;
```

Comment:
Not sure about this because of what we discussed privately.

Reply:
I did this because I thought I had to do this in order to differentiate between different configurations, but I realized this is always done automatically...